src/App.tsx

```diff
@@ -1,26 +1,57 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/header/Header';
+import { TodoList } from './components/todoList/TodoList';
+import { Footer } from './components/footer/Footer';
+import { Errors } from './components/errors/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState('all');
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        switch (status) {
+          case 'all':
+            setTodos(todosFromServer);
+            break;
+          case 'active':
+            setTodos(todosFromServer.filter(todo => !todo.completed));
+            break;
+          case 'completed':
+            setTodos(todosFromServer.filter(todo => todo.completed));
+            break;
+          default:
+            setTodos(todosFromServer);
```

Comment:
It needs to be some sort of filter function outside the component you need to fetch todos only one time when your application is first time rendered